python-lib/gluonts_forecasts/mxnet_utils.py

```diff
@@ -0,0 +1,47 @@
+try:
+    import mxnet as mx
+except OSError as cuda_error:  # error when importing mxnet
+    raise Exception(f"Error when importing mxnet: {cuda_error}")
+
+from constants import GPU_CONFIGURATION
+
+
+class GPUError(Exception):
+    """Custom exception raised when the GPU selection failed"""
+
+    pass
+
+
+def set_mxnet_context(gpu_devices):
+    """Return the right MXNet context from the selected GPU configuration.
+
+    Args:
+        gpu_devices (list): List of gpu device numbers or 'container_gpu'. Default to None which means no gpu.
+
+    Raises:
+        GPUError:
+            If mx.context.num_gpus() fails on a Cuda error.
+            If no GPUs are detected on the DSS server or in the container.
+
+    Returns:
+        A mxnet.context.Context to use for Deep Learning models training.
+    """
+
+    if gpu_devices is None:
+        return mx.context.cpu()
+    else:
+        try:
+            num_gpu = mx.context.num_gpus()
+        except mx.base.MXNetError as num_gpus_error:  # error on num_gpus()
+            raise GPUError(f"Cuda error: {num_gpus_error}")
+
+        if num_gpu == 0:
+            if GPU_CONFIGURATION.CONTAINER_GPU in gpu_devices:
```

Comment:
That's no longer valid since we changed the UX, right?

Reply:
gpu_devices is either [$GPU_NUMBER] or ["container_gpu"]

Reply:
ok, closing